### ISSUE TITLE
[Bug]: `createdAt` key for author not found

### ISSUE THREAD

**Issue by rhysm94** [MasterJ93/ATProtoKit#49]
### Summary

When trying to pull the timeline with my own post, I'm getting a decoding error at `post.author.createdAt`

### Reproduction Steps

I posted and then tried to access my timeline via ATProtoKit.
Instead, I got a decoding failure at `post.author.createdAt`.
I don't know if this is an issue with my account, or if it's a problem with access your own posts?

### Expected Results

I expected fetching the timeline to succeed.

### Actual Results

I got the decoding error.

### What operating systems did you experience this bug? (We'll count Docker as an operating system.)

Other

### Operating System Version

iOS 18.2 beta 3

### ATProtoKit Version

0.19.2

### Additional Context

Looking through the timeline lexicon, it seems that the field is optional, but I can see that the way the `PostViewDefinition` is written, the `createdAt` field is optional and has a special `Codable` wrapper.

Here's the error as dumped in Xcode:

```swift
.failure(
  .keyNotFound(
	.CodingKeys(stringValue: "createdAt", intValue: nil),
	DecodingError.Context(
	  codingPath: [
		[0]: .CodingKeys(stringValue: "feed", intValue: nil),
		[1]: .index(0),
		[2]: .CodingKeys(stringValue: "post", intValue: nil),
		[3]: .CodingKeys(stringValue: "author", intValue: nil)
	  ],
	  debugDescription: #"No value associated with key CodingKeys(stringValue: "createdAt", intValue: nil) ("createdAt")."#,
	  underlyingError: nil
	)
  )
)
```

and this is the JSON that triggered it:

```json
"post": {
	"uri": "at://did:plc:eltnk24yvefmic5qtzb35zpu/app.bsky.feed.post/3lbd6psqh4s2u",
	"cid": "bafyreigvavncnnbde6tjnrybbq5gb2rg2kr5urgxgpn6hcqilwhtafumna",
	"likeCount": 0,
	"replyCount": 0,
	"repostCount": 0,
	"quoteCount": 0,
	"author": {
	  "did": "did:plc:eltnk24yvefmic5qtzb35zpu",
	  "handle": "rhysm.wales",
	  "displayName": "Rhys Morgan",
	  "avatar": "https://cdn.bsky.app/img/avatar/plain/did:plc:eltnk24yvefmic5qtzb35zpu/bafkreidza733ymr2ozrru3vtczkzpbfstc6o22vcczlte5w5mwt4p755t4@jpeg"
	},
	"record": {
	  "text": "Is there *any* kind of async alternative to a CurrentValueSubject yet? Is it something I have to make myself? ...can I even make something like that safely???",
	  "$type": "app.bsky.feed.post",
	  "langs": [
		"en"
	  ],
	  "reply": {
		"root": {
		  "cid": "bafyreihaqs6ddh5jhe2rxmqfzmg3dfjn2327ywbz3v3llebulwwim6gpme",
		  "uri": "at://did:plc:klsh7edzj3jmxucibyjqstb3/app.bsky.feed.post/3lbcicg64u22j"
		},
		"parent": {
		  "cid": "bafyreihaqs6ddh5jhe2rxmqfzmg3dfjn2327ywbz3v3llebulwwim6gpme",
		  "uri": "at://did:plc:klsh7edzj3jmxucibyjqstb3/app.bsky.feed.post/3lbcicg64u22j"
		}
	  },
	  "createdAt": "2024-11-19T19:37:56.903Z"
	},
	"indexedAt": "2024-11-19T19:37:57.466Z"
}
```

**Comment by MasterJ93** [comment]
Thanks for the report. I'm in progress at looking into it. It will be a 0.20.x fix.

**Comment by MasterJ93** [comment]
The `develop1 branch has been updated with a fix to this issue. If you're still experiencing an error with respect to this, it's possible I simply haven't gotten to that part of the code yet. The fix is that I've changed the way how ATProtoKit detects and handles the conversion between `Date[?]` and the ISO8601 formatted string. I have been awake for... quite some time, so I need to sleep, but I'll finish things up later today. Hopefully I can final put it into the `main` branch and tag it within the next 48 hours.
